packages/hooks/src/__tests__/useResidenceSelfDeclaration.spec.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { renderHook } from '@testing-library/react-hooks';
+import useResidenceSelfDeclaration from '../useResidenceSelfDeclaration';
+
+describe('useResidenceSelfDeclaration', () => {
+    test('should return true if client residence is spain', async () => {
+        const mock = mockStore({
+            client: {
+                residence: 'es',
+                residence_list: [
+                    {
+                        value: 'es',
+                        account_opening_signup_declaration_required: true,
+                    },
+                    {
+                        value: 'id',
+                        account_opening_signup_declaration_required: false,
+                    },
+                ],
+            },
+        });
+
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock}>{children}</StoreProvider>
+        );
+        const { result } = renderHook(() => useResidenceSelfDeclaration(), { wrapper });
+        expect(result.current.is_residence_self_declaration_required).toBe(true);
+    });
+
+    test('should return false if client residence is not spain', async () => {
```

Comment:
Something like:
test(' should return false if client's residence does not require self-declaration', async ()

Reply:
this too is done 👍🏼